### PR TITLE
added loop, conditional, and variable blocks

### PR DESCRIPTION
**Pull request by mpoegel**
#9

### CODE REVIEW

File: app/js/block.js [hunk 1]
```diff
@@ -90,3 +119,88 @@ Block.dragStart = (event) => {
...
+    this.variable = null;
+    this.onTrue = null;
+    this.onFalse = null;
+  }
+
+  getParamsMeta() {
+    return `Block ${this.uid}: conditionaaal`;
+  }
+
+  toJSON() {
```
Comment (droghio) [inline app/js/block.js]: Nice, this should make saving/loading projects much simpler.

File: app/js/block.js [hunk 2]
```diff
@@ -90,3 +119,88 @@ Block.dragStart = (event) => {
...
+  constructor(x, y) {
+    super(x, y);
+    this.value = null;
+  }
+
+  getParamsMeta() {
+    return `Block ${this.uid}: variable?!`;
+  }
+
+  toJSON() {
```
Comment (droghio) [inline app/js/block.js]: Hmm, we seem to be repeating this alot. Maybe we should have a generic toJSON in the main block class and add to the object is necessary in each child.

File: app/js/ui.js [hunk 3]
```diff
@@ -38,10 +44,60 @@ class Controller {
...
+    this.toggleConfigBar();
+    selectBlock();
+  }
+
+  toggleConfigBar() {
+    const configBar = document.getElementById('config-bar');
+    const configContent = document.getElementById('config-bar').getElementsByTagName('content')[0];
+    const mainWindow = document.getElementsByTagName('main')[0];
+    const viewportWidth = document.documentElement.clientWidth;
+    if (this.configBarActive) {
```
Comment (droghio) [inline app/js/ui.js]: My preferred way to handle style updates like this is to assign a class to the parent. I.e. give the body a "sidebar" or "blockbar" class and keeping style changes in the css.

File: app/js/block.js (hunk 2 above)
Comment (mpoegel) [inline app/js/block.js]: Yeah I thought about that. It's probably a good idea because we will have so many block subclasses.

File: app/js/ui.js [hunk 4]
```diff
@@ -38,10 +44,60 @@ class Controller {
...
-      const block = blocks[i];
-      this.templateBlocks.push(new TemplateBlock(block));
+    const templateBlocks = document.querySelectorAll('.block.template');
+    for (let i = 0; i < templateBlocks.length; i++) {
+      const block = templateBlocks[i];
+      this.templateBlocks.push(new blocks.TemplateBlock(block));
+    }
+  }
+
+  createBlock(blockType, x, y) {
```
Comment (mpoegel) [inline app/js/ui.js]: I tried to make it as easy as possible to add blocks when loading them from a file.

**Pull request merged**